### PR TITLE
Represent symlink actions in the compact execution log

### PR DESCRIPTION
**Pull request by fmeum**
Symlink actions create a scheduling and invalidation dependency between their input and output artifact, but so far have not been represented in the execution log as they aren't backed by a spawn.

This change adds a new `SymlinkAction` message to the compact execution log that represents these actions. The legacy execution log format isn't modified.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/exec/ExpandedSpawnLogContext.java [hunk 1]
```diff
@@ -282,6 +283,11 @@ public void logSpawn(
     }
   }
 
+  @Override
+  public void logSymlinkAction(AbstractAction action) {
+    // The expanded log does not report symlink actions.
```
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/exec/ExpandedSpawnLogContext.java:288]: I could represent these actions as synthetic spawns.

Comment (tjgq) [reply src/main/java/com/google/devtools/build/lib/exec/ExpandedSpawnLogContext.java:288]: I'm fine with not implementing it unless someone requests it.

File: src/main/protobuf/failure_details.proto [hunk 2]
```diff
@@ -1148,6 +1148,7 @@ message SymlinkAction {
     EXECUTABLE_INPUT_CHECK_IO_EXCEPTION = 3 [(metadata) = { exit_code: 1 }];
     LINK_CREATION_IO_EXCEPTION = 4 [(metadata) = { exit_code: 1 }];
     LINK_TOUCH_IO_EXCEPTION = 5 [(metadata) = { exit_code: 1 }];
+    LINK_LOG_IO_EXCEPTION = 6 [(metadata) = { exit_code: 1 }];
```
Comment (tjgq) [inline src/main/protobuf/failure_details.proto:1151]: Let's use 36 here.

**Review by tjgq — CHANGES_REQUESTED** [review]

**Review by tjgq — APPROVED** [review]

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Pull request closed without merging**